### ISSUE TITLE
Failed to create file appender on UWP

### ISSUE THREAD

**Issue by sakatuba** [NLog/NLog#3059]
Hi!

I found a little bug on FileTarget.

**NLog version**: 4.5.11

**Platform**: UniversalWindows (1803/17134, Windows10 64bit)

**Current NLog config**: I did it by programable way just below.

```C#
            FileTarget fileTarget = new FileTarget("fileTarget")
            {
                Encoding = Encoding.UTF8,
                Layout = "${date:format=yyyy-MM-dd HH\\:mm\\:ss.fff}\t${level}\t${logger}\t${message}\t${exception:format=toString,Data:maxInnerExceptionLevel=10}",
                FileName = @"${var:LogPath}\XXX\log-${date:format=yyyy-MM-dd}.log",
                WriteBom = false,
                LineEnding = LineEndingMode.CRLF,
                KeepFileOpen = true,
                ConcurrentWrites = true,
                OptimizeBufferReuse = true,
                CreateDirs = true
            };
            config.AddTarget(fileTarget);
```

- What is the current result?

No Log is logged. After making InnerLogger on, I found exceptions.

- What is the expected result?

Make it run and log.

- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?

Yes, I did.

- Please post full exception details (message, stacktrace, inner exceptions)

```
NLog: 2018-12-28 10:18:40.1169 Warn Failed to create file appender: C:\Users\sakamoto\AppData\Local\Packages\~~~~~~~\LocalState\log-2018-12-28.log Exception: System.UnauthorizedAccessException: Access to the path 'Global\NLog-FileFileLock-c:_users_sakamoto_appdata_local_packages_~~~~~_localstate_log-2018-12-28.log' is denied.
   at System.IO.__Error.WinIOError(Int32 errorCode, String str)
   at System.Threading.Mutex.MutexTryCodeHelper.MutexTryCode(Object userData)
   at System.Runtime.CompilerServices.RuntimeHelpers.ExecuteCodeWithGuaranteedCleanup(TryCode code, CleanupCode backoutCode, Object userData)
   at System.Threading.Mutex.CreateMutexWithGuaranteedCleanup(Boolean initiallyOwned, String name, Boolean& createdNew, SECURITY_ATTRIBUTES secAttrs)
   at System.Threading.Mutex..ctor(Boolean initiallyOwned, String name, Boolean& createdNew, MutexSecurity mutexSecurity)
   at NLog.Internal.FileAppenders.BaseMutexFileAppender.CreateSharableMutex(String mutexNamePrefix)
   at NLog.Internal.FileAppenders.MutexMultiProcessFileAppender..ctor(String fileName, ICreateFileParameters parameters)
   at NLog.Internal.FileAppenders.MutexMultiProcessFileAppender.Factory.NLog.Internal.FileAppenders.IFileAppenderFactory.Open(String fileName, ICreateFileParameters parameters)
   at NLog.Internal.FileAppenders.FileAppenderCache.CreateAppender(String fileName, Int32 freeSpot)

```

- Are there any workarounds? yes

After few hours of try-and error,
Just turn off 'ConcurrentWrites', exception is gone and logs are logged.

And I read source code of BaseMutexFileAppender.cs, I fownd some strings that appears in inner target log.
https://github.com/NLog/NLog/blob/dev/src/NLog/Internal/FileAppenders/BaseMutexFileAppender.cs

```C#
        private string GetMutexName(string mutexNamePrefix)
        {
            const string mutexNameFormatString = @"Global\NLog-File{0}-{1}";
            const int maxMutexNameLength = 260;
```

As far as my concerned, on UWP, Mutex path must be on permitted directory path.
But under 'ConcurrentWrites = true' configuration, it is not configuable. 
And not permitted (C:\)Global\NLog-File~ to write.

- Is there a version in which it did work? 

I have no idea.

- Can you help us by writing an unit test?

Sorry, I can't.

I hope this poor english report may be understood.

**Comment by snakefoot** [comment]
Glad you found the issue. Have create a PR for changing the default value of ConcurrentWrites for NetStandard: #3048

**Comment by 304NotModified** [comment]
> After few hours of try-and error,
> Just turn off 'ConcurrentWrites', exception is gone and logs are logged.

As a small recap, this isn't working in UWP due to limitations.

**Comment by sakatuba** [comment]
Happy new year !

Rolf-san Julian-san, Thank you very much.

best regards.

**Comment by 304NotModified** [comment]
resolved with #3079

**Comment by 304NotModified** [comment]
FYI, NLog 4.6 is online which includes a fix fir this
